dlt/destinations/impl/filesystem/filesystem.py

```diff
@@ -220,3 +239,126 @@ def __exit__(
 
     def should_load_data_to_staging_dataset(self, table: TTableSchema) -> bool:
         return False
+
+    #
+    # state stuff
+    #
+
+    def _write_to_json_file(self, filepath: str, data: DictStrAny) -> None:
+        dirname = os.path.dirname(filepath)
+        if not self.fs_client.isdir(dirname):
```

Comment:
there is a test that fails because _write_to_json_file will be called without an initialised dataset. I can't quite remember which one, but i will have a look.